underfs/cos/src/main/java/alluxio/underfs/cos/COSInputStream.java

```diff
@@ -81,7 +96,22 @@ protected InputStream createStream(long startPos, long endPos)
     GetObjectRequest req = new GetObjectRequest(mBucketName, mKey);
     // COS returns entire object if we read past the end
     req.setRange(startPos, endPos < mContentLength ? endPos - 1 : mContentLength - 1);
-    COSObject object = mCosClient.getObject(req);
-    return new BufferedInputStream(object.getObjectContent());
+    CosServiceException lastException = null;
+    while (mRetryPolicy.attempt()) {
+      try {
+        COSObject object = mCosClient.getObject(req);
+        return new BufferedInputStream(object.getObjectContent());
+      } catch (CosServiceException e) {
+        LOG.warn("Attempt {} to open key {} in bucket {} failed with exception : {}",
+            mRetryPolicy.getAttemptCount(), mKey, mBucketName, e.toString());
+        if (e.getStatusCode() != HttpStatus.SC_NOT_FOUND) {
+          throw e;
```

Comment:
`CosServiceException` is a `RuntimeException`. wrap the thrown exception in an IOException

Reply:
Wrapped and threw, thanks